x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/token/TransportCreateTokenAction.java

```diff
@@ -86,19 +85,15 @@ private void authenticateAndCreateToken(CreateTokenRequest request, ActionListen
     }
 
     private void createToken(CreateTokenRequest request, Authentication authentication, Authentication originatingAuth,
-                             boolean includeRefreshToken, ActionListener<CreateTokenResponse> listener) {
-        try {
-            tokenService.createUserToken(authentication, originatingAuth, ActionListener.wrap(tuple -> {
-                final String tokenStr = tokenService.getAccessTokenAsString(tuple.v1());
-                final String scope = getResponseScopeValue(request.getScope());
-
-                final CreateTokenResponse response =
-                    new CreateTokenResponse(tokenStr, tokenService.getExpirationDelay(), scope, tuple.v2());
-                listener.onResponse(response);
-            }, listener::onFailure), Collections.emptyMap(), includeRefreshToken);
-        } catch (IOException e) {
-            listener.onFailure(e);
-        }
+            boolean includeRefreshToken, ActionListener<CreateTokenResponse> listener) {
+        tokenService.createUserToken(authentication, originatingAuth, Collections.emptyMap(), includeRefreshToken,
```

Comment:
"user token" indeed feels more like an internal thing that holds the user details associated with a security token. I don't have any preference for an alternative.

Reply:
I recently changed the `getUserTokenString` method to `getAccessTokenAsString` . The use of `UserToken`(sometimes) interchangeably with `accessToken` throughout, is a remnant of the past where the the serialized user token was indeed the access token and it is one of the things that added to the complexity of the TokenService. I agree we should change this where necessary. I propose to use `UserToken` when dealing with our internal structure, `AccessToken` and `RefreshToken` respectively where needed, and finally `OAuth2Tokens` when we need to refer to both tokens. 

I think it fits in this effort, but we can do it in a separate PR to keep scope tight.

Reply:
So, `createOAuth2Tokens` since this method returns both `access` and `refresh`, right? I will rename, the scope of this is broad and it's easy to ignore during review.